From adc0b01f990f8e19c6027aa6bed0ddfbfe54db28 Mon Sep 17 00:00:00 2001
From: Matthias Schiffer <matthias.schiffer@tq-group.com>
Date: Mon, 2 Sep 2024 13:37:00 +0200
Subject: [PATCH] imx8m-hantro: dummy_encoder: implement
 imx_vpu_api_enc_get_encoded_frame_ext()

Linking gstreamer-imx will fail if imx_vpu_api_enc_get_encoded_frame_ext()
is not available.

Signed-off-by: Matthias Schiffer <matthias.schiffer@tq-group.com>
Upstream-Status: Submitted [https://github.com/Freescale/libimxvpuapi/pull/62]
---
 imxvpuapi2/imxvpuapi2_imx8m_hantro_dummy_encoder.c | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/imxvpuapi2/imxvpuapi2_imx8m_hantro_dummy_encoder.c b/imxvpuapi2/imxvpuapi2_imx8m_hantro_dummy_encoder.c
index 1568add..8a88a34 100644
--- a/imxvpuapi2/imxvpuapi2_imx8m_hantro_dummy_encoder.c
+++ b/imxvpuapi2/imxvpuapi2_imx8m_hantro_dummy_encoder.c
@@ -119,11 +119,16 @@ ImxVpuApiEncReturnCodes imx_vpu_api_enc_encode(ImxVpuApiEncoder *encoder, size_t
 	return IMX_VPU_API_ENC_RETURN_CODE_OK;
 }
 
-
 ImxVpuApiEncReturnCodes imx_vpu_api_enc_get_encoded_frame(ImxVpuApiEncoder *encoder, ImxVpuApiEncodedFrame *encoded_frame)
+{
+	return imx_vpu_api_enc_get_encoded_frame_ext(encoder, encoded_frame, NULL);
+}
+
+ImxVpuApiEncReturnCodes imx_vpu_api_enc_get_encoded_frame_ext(ImxVpuApiEncoder *encoder, ImxVpuApiEncodedFrame *encoded_frame, int *is_sync_point)
 {
 	IMX_VPU_API_UNUSED_PARAM(encoder);
 	IMX_VPU_API_UNUSED_PARAM(encoded_frame);
+	IMX_VPU_API_UNUSED_PARAM(is_sync_point);
 	return IMX_VPU_API_ENC_RETURN_CODE_OK;
 }
 
